modules/tls/src/main/java/org/mule/runtime/module/tls/internal/DefaultTlsContextFactory.java

```diff
@@ -213,12 +218,24 @@ public SSLContext createSslContext() throws KeyManagementException, NoSuchAlgori
   }
 
   @Override
+  public SSLSocketFactory getSocketFactory() throws KeyManagementException, NoSuchAlgorithmException {
+    return new RestrictedSSLSocketFactory(createSslContext(), getEnabledCipherSuites(), getEnabledProtocols());
+  }
+
+  @Override
+  public SSLServerSocketFactory getServerSocketFactory() throws KeyManagementException, NoSuchAlgorithmException {
+    return new RestrictedSSLServerSocketFactory(createSslContext(), getEnabledCipherSuites(), getEnabledProtocols());
+  }
+
+  @Override
   public String[] getEnabledCipherSuites() {
-    if (isUseDefaults(enabledCipherSuites)) {
-      return tlsConfiguration.getEnabledCipherSuites();
+    String[] enabledCipherSuites;
+    if (isUseDefaults(this.enabledCipherSuites)) {
```

Comment:
Yes, there's a check on initialise() where (if set) the protocols and ciphers are intersected with the global ones to check for differences, failing if that is the case.